addons/binding/org.openhab.binding.icloud/README.md

```diff
@@ -0,0 +1,77 @@
+
+Please note: Application specific passwords are not supported.
+You´ll need to provide the account password to the binding.
+
+## Discovery
+
+An iCloud account can be added as a thing and needs to be configured with your ID and password.
+The devices registered to this account will then be automatically discovered.
+
+## Binding Configuration
+
+The Binding has no configuration options, all configuration is done at Thing level.
+
+## Thing Configuration
+
+The Binding provides two Thing types.
+
+### iCloud Account (Bridge) Thing
+
+The following table lists the configuration parameters:
+
+| Parameter            | Description |                             |
+|----------------------|-------------|-----------------------------|
+| appleId              |             | mandatory                   |
+| password             |             | mandatory                   |
+| refreshTimeInMinutes |             | optional, 5 minutes default |
+
+A device is identified by a hash value calculated from the device id provided by apple.
+If a device is removed from the account the respective openHAB Thing will go OFFLINE.
+
+## Channels
+
+### Device
+
+The following channels are available (if supported by the device):
+
+| Channel ID         | Type     | Description                                                                                                                                 |
+|--------------------|----------|---------------------------------------------------------------------------------------------------------------------------------------------|
+| batteryStatus      | String   | Current battery status (Charging, Charged, Unknown)                                                                                         |
+| batteryLevel       | Number   | Battery charge in %                                                                                                                         |
+| findMyPhone        | Switch   | Triggers the ["Find my phone"](https://support.apple.com/explore/find-my-iphone-ipad-mac-watch) functionality of the device (if available). |
+| location           | Location | Location of the device.                                                                                                                     |
+| locationAccuracy   | Number   | Accuracy of the last position report.                                                                                                       |
+| locationLastUpdate | DateTime | Timestamp of the last location update.                                                                                                      |
+
+## Full Example
+
+### icloud.things
+
+```php
+Bridge icloud:account:myaccount [appleId="abc@xyz.tld", password="secure", refreshTimeInMinutes=5]
+{
+    Thing device myiPhone8 "iPhone 8" @ "World" [deviceId="VIRG9FsrvXfE90ewVBA1H5swtwEQePdXVjHq3Si6pdJY2Cjro8QlreHYVGSUzuWV"]
+}
+```
+
+The "label" @ "location" part is optional (as always).
```

Comment:
I was the one to suggest this sentence. I believe you can remove it. In an example everything is subject to modification.